cpp/src/DataStorm/ForwarderManager.h

```diff
@@ -13,19 +13,15 @@
 
 namespace DataStormI
 {
-
-    class Instance;
-
     class ForwarderManager : public Ice::BlobjectAsync
     {
     public:
         using Response = std::function<void(bool, const Ice::ByteSeq&)>;
         using Exception = std::function<void(std::exception_ptr)>;
 
         ForwarderManager(const Ice::ObjectAdapterPtr&, const std::string&);
-
-        std::optional<Ice::ObjectPrx> add(std::function<void(Ice::ByteSeq, Response, Exception, const Ice::Current&)>);
-        std::optional<Ice::ObjectPrx> add(std::function<void(Ice::ByteSeq, const Ice::Current&)>);
+        Ice::ObjectPrx add(std::function<void(Ice::ByteSeq, Response, Exception, const Ice::Current&)>);
+        Ice::ObjectPrx add(std::function<void(Ice::ByteSeq, const Ice::Current&)>);
```

Comment:
Maybe we should make this generics to avoid the uncheckedCasts?

Reply:
Yes, I would convert them into template functions.